src/appservice/AppService.ts

```diff
@@ -114,10 +116,12 @@ export class MjolnirAppService {
         // Acts as an alternative to the web api provided for the widget.
         if ('m.room.member' === mxEvent.type) {
             if ('invite' === mxEvent.content['membership'] && mxEvent.state_key === this.bridge.botUserId) {
-                await this.mjolnirManager.provisionNewMjolnir(mxEvent.sender);
+                await this.mjolnirManager.provisionNewMjolnir(mxEvent.sender).catch((e: any) => {
```

Comment:
Let's not mix and match two syntaxes.

```js
try {
  await ...
} catch (e) {
  log.error(...);
  // `return`, I guess?
} ```

And the same below.

Reply:
Hmmm, I'm tempted to disagree, they're both good in different contexts + `finally` is needed in some situations.
Plus we don't need to return in this case because we still want to reject their invite if we can't provision a mjolnir.
We don't use `try` in this file either.  Though I think the observation that elsewhere in mjolnir we tend to use `try` would probably be correct.

Reply:
I think you could be right that `try` makes it a bit more explicit that we are handling errors and continuing though, so i will change it

Reply:
```
                try {
                    await this.mjolnirManager.provisionNewMjolnir(mxEvent.sender)
                } catch (e: any) {
                    log.error(`Failed to provision a mjolnir for ${mxEvent.sender} after they invited ${this.bridge.botUserId}:`, e);
                    // continue, we still want to reject this invitation.
                }
                try {
                    // reject the invite to keep the room clean and make sure the invetee doesn't get confused and think this is their mjolnir.
                    await this.bridge.getBot().getClient().leaveRoom(mxEvent.room_id);
                } catch (e: any) {
                    
                    log.warn("Unable to reject an invite to a room", e);
                }
  ```
  
  Naa, this is noisy and ugly